app/src/pencil2d.cpp

```diff
@@ -0,0 +1,130 @@
+Copyright (C) 2005-2007 Patrick Corrieri & Pascal Naidon
+Copyright (C) 2012-2020 Matthew Chiawen Chang
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+*/
+
+#include <memory>
+
+#include <QDebug>
+#include <QFileOpenEvent>
+#include <QIcon>
+#include <QLibraryInfo>
+#include <QSettings>
+#include <QTranslator>
+
+#include "commandlineexporter.h"
+#include "commandlineparser.h"
+#include "mainwindow2.h"
+#include "pencildef.h"
+#include "platformhandler.h"
+
+#include "pencil2d.h"
+
+Pencil2D::Pencil2D(int& argc, char** argv) :
+    QApplication(argc, argv)
+{
+    // Set organization and application name
+    setOrganizationName("Pencil2D");
+    setOrganizationDomain("pencil2d.org");
+    setApplicationName("Pencil2D");
+    setApplicationDisplayName("Pencil2D");
+
+    // Set application version
+    setApplicationVersion(APP_VERSION);
+
+    // Set application icon
+    setWindowIcon(QIcon(":/icons/icon.png"));
+
+    // Associate the application with our desktop entry
+    setDesktopFileName("org.pencil2d.Pencil2D.desktop");
+
+    installTranslators();
+}
+
+Status Pencil2D::handleCommandLineOptions()
+{
+    CommandLineParser parser;
+    parser.process(arguments());
+
+    QString inputPath = parser.inputPath();
+    QStringList outputPaths = parser.outputPaths();
+
+    if (outputPaths.isEmpty())
+    {
+        prepareGuiStartup(inputPath);
+        return Status::OK;
+    }
+
+    auto mainWindow = new MainWindow2;
+    CommandLineExporter exporter(mainWindow);
+    if (exporter.process(inputPath,
+                         outputPaths,
+                         parser.camera(),
+                         parser.width(),
+                         parser.height(),
+                         parser.startFrame(),
+                         parser.endFrame(),
+                         parser.transparency()))
+    {
+        return Status::SAFE;
+    }
+    return Status::FAIL;
+}
+
+bool Pencil2D::event(QEvent* event)
+{
+    if (event->type() == QEvent::FileOpen)
+    {
+        auto fileOpenEvent = dynamic_cast<QFileOpenEvent*>(event);
+        Q_ASSERT(fileOpenEvent);
+        emit openFileRequested(fileOpenEvent->file());
+        return true;
+    }
+    return QApplication::event(event);
+}
+
+void Pencil2D::installTranslators()
+{
+    QSettings setting(PENCIL2D, PENCIL2D);
+    QString userLocale = setting.value(SETTING_LANGUAGE).toString();
+    QLocale locale = userLocale.isEmpty() ? QLocale::system() : QLocale(userLocale);
+    QLocale::setDefault(locale);
+
+    std::unique_ptr<QTranslator> qtTranslator(new QTranslator(this));
```

Comment:
Should a QTranslator be deleted after levaing the function? 
Also two memory management methods are mixed here while both `std::unique_ptr` and parent pointer are set at the same time. Usually only one is needed.

Reply:
Had a look at the QTranslator source code. A plain `QTranslator* translator = new QTranslator(this);` is good. `std::unique_ptr` is not necessary.

QTranslator calls `QCoreApplication::removeTranslator(this);` in the destructor, so we probably want to keep QTranslator alive rather than deleting in right away. Just leave the object deletion to its parent object.
https://code.woboq.org/qt5/qtbase/src/corelib/kernel/qtranslator.cpp.html#_ZN11QTranslatorD1Ev

Reply:
oh sorry, didn't notice that you use `qtTranslator.release()` to give away the ownership. Then it's ok now, My bad.